android/capacitor/src/main/java/com/getcapacitor/plugin/util/HttpRequestHandler.java

```diff
@@ -398,6 +398,23 @@ public static JSObject request(PluginCall call, String httpMethod, Bridge bridge
 
         boolean isHttpMutate = method.equals("DELETE") || method.equals("PATCH") || method.equals("POST") || method.equals("PUT");
 
+        // a workaround for the following android web view issue:
+        // https://issues.chromium.org/issues/40450316
+        // x-cap-user-agent contains the user agent set in JavaScript
+        if (headers.has("x-cap-user-agent")) {
+            String userAgentValue = headers.getString("x-cap-user-agent");
+            if (userAgentValue != null) {
+                headers.put("User-Agent", userAgentValue);
+            }
+
+            headers.remove("x-cap-user-agent");
+        }
```

Comment:
Same as the previous comment, also, the previous one was checking for `isEmpty()` too, so we should probably do the same checks
```suggestion
       String userAgentValue = headers.getString("x-cap-user-agent");
        if (userAgentValue != null && !userAgentValue.isEmpty()) {
            headers.put("User-Agent", userAgentValue);
            headers.remove("x-cap-user-agent");
        }
```